harvester/GeoNetworkRepository.py

```diff
@@ -0,0 +1,329 @@
+            language = find_ns(find_ns(geonetwork_record, "gmd:language"), "gmd:LanguageCode").attrib["codeListValue"]
+        elif get_gco_CharacterString(find_ns(geonetwork_record, "gmd:language")) is not None:
+            language = get_gco_CharacterString(find_ns(geonetwork_record, "gmd:language"))
+
+        if language == "fre":
+            record["title_fr"] = record["title"]
+            record["title"] = ""
+            record["tags_fr"] = record["tags"]
+            record["tags"] = []
+            record["subject_fr"] = record["subject"]
+            record["subject"] = []
+            record["description_fr"] = record["description"]
+            record["description"] = ""
+
+
+        # Geospatial
+        extents = findall_ns(data_identification, "gmd:extent")
+        if extents is not None:
+            record["geobboxes"] = []
+            record["geoplaces"] = []
+            for extent in extents:
+                ex_extent = find_ns(extent, "gmd:EX_Extent")
+                geographicElements = findall_ns(ex_extent, "gmd:geographicElement")
+                # Geographic extent (ignore temporal)
+                if geographicElements is not None:
+                    record["geoplaces"].append({"place_name": get_gco_CharacterString(find_ns(ex_extent, "gmd:description"))})
+                    # Geospatial - bounding boxes only, not polygons
+                    for geographicElement in geographicElements:
+                        geographicBoundingBox = find_ns(geographicElement, "gmd:EX_GeographicBoundingBox")
+                        record["geobboxes"].append({"westLon": find_ns(find_ns(geographicBoundingBox,"gmd:westBoundLongitude"), "gco:Decimal").text,
+                                                    "eastLon": find_ns(find_ns(geographicBoundingBox, "gmd:eastBoundLongitude"),"gco:Decimal").text,
+                                                    "northLat": find_ns(find_ns(geographicBoundingBox, "gmd:northBoundLatitude"), "gco:Decimal").text,
+                                                    "southLat": find_ns(find_ns(geographicBoundingBox, "gmd:southBoundLatitude"), "gco:Decimal").text})
+
+        return record
+
+    @rate_limited(5)
+    def _update_record(self, record):
+        request_success = False
+        request_count = 0
+        while not request_success and request_count < 5:
+            try:
+                xml_record_url = "https://hecate.hakai.org/geonetwork/srv/api/records/{}/formatters/xml".format(record["local_identifier"])
+                response = requests.request("GET", xml_record_url)
+                if response.status_code == 200:
+                    request_success = True
+                elif response.status_code == 400:
+                    request_count += 1
+                    self.logger.info("Trying again to fetch record {}: Response {}".format(record["local_identifier"], response.status_code))
```

Comment:
@prairiewest similar question as above - basically after a few tries (usually on the 2nd or 3rd), I'll get a 200, but there are occasionally a few glitches first.

Reply:
The "not 200" codes may be due to some caching on the server side that has to retrieve data, and can't get it done within the required time... but on later requests, the data is finally available and the request succeeds.  So sleeping longer may cut down on the number of error requests.

Reply:
Thanks, this is helpful! I've increased the sleep time.